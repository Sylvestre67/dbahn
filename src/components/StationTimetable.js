import React from 'react';
// import PropTypes from 'prop-types';
import { Query, withApollo } from 'react-apollo';
import { withRouter } from 'react-router-dom';

import { withStyles } from '@material-ui/core/styles';

import LinearProgress from '@material-ui/core/LinearProgress';
import Typography from '@material-ui/core/Typography';

import StationFragments from '../context/fragments';
import { GET_STATION } from '../context/queries';

const styles = theme => ({
	root: {},
});

const StationTimetable = props => {
	const { classes, client, match } = props;
	let timetable;

	try {
		timetable = client.readFragment({
			id: `Station_${match.params.evaId}`,
			fragment: StationFragments.fragments.timetable,
		});
	} catch (error) {
		timetable = null;
	}

	return Object.is(timetable, null) ? (
		<Query
			query={GET_STATION}
			variables={{ evaId: match.params.evaId }}
			client={client}>
			{({ data, loading, error }) => (
				<React.Fragment>
					{loading ? (
						<LinearProgress color="secondary" />
					) : (
						<div className={classes.root}>
							<Typography variant="h5">Timetable</Typography>
							<Typography>
								{JSON.stringify(
									data.stationWithEvaId.timetable,
									4,
									4
								)}
							</Typography>
						</div>
					)}
				</React.Fragment>
			)}
		</Query>
	) : (
		<div className={classes.root}>
			<Typography variant="h5">Timetable</Typography>
			<Typography>{JSON.stringify(timetable, 4, 4)}</Typography>
		</div>
	);
};

export default withStyles(styles)(withRouter(withApollo(StationTimetable)));
