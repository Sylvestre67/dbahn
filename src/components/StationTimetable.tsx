import React from 'react';
import { ApolloClient } from 'apollo-client';

import { withRouter, RouteComponentProps } from 'react-router-dom';
import { withStyles, createStyles, WithStyles } from '@material-ui/core/styles';

import StationFragments from '../context/fragments';
import { GET_STATION } from '../context/queries';

import { Timetable } from '../context/__generated__/Timetable';

import StationQuery from './StationQuery';
import TimeTableViz from './TimeTableViz';

interface IStationTimetableProps extends WithStyles<typeof styles> {}
interface IStationTimetableProps
	extends RouteComponentProps<{ evaId: string }> {
	client: ApolloClient<any>;
}

const styles = () =>
	createStyles({
		root: { flex: 1, backgroundColor: '#eee' },
	});

const StationTimetable: React.SFC<IStationTimetableProps> = props => {
	const { client, match } = props;

	// Querying the cache, if timetable fragment found, use it, do not query.
	let timetable: Timetable | null;
	try {
		timetable = client.readFragment({
			id: `Station_${match.params.evaId}`,
			fragment: StationFragments.fragments.timetable,
		});
	} catch (error) {
		timetable = null;
	}

	return Object.is(timetable, null) ? (
		<StationQuery
			query={GET_STATION}
			variables={{ evaId: parseInt(match.params.evaId, 10) }}
			client={client}>
			{({ data, loading, error }) => (
				<TimeTableViz data={data} loading={loading} error={error} />
			)}
		</StationQuery>
	) : (
		<TimeTableViz
			data={client.readQuery({
				query: GET_STATION,
				variables: { evaId: parseInt(match.params.evaId, 10) },
			})}
			loading={false}
			error={undefined}
		/>
	);
};

export default withStyles(styles)(withRouter(StationTimetable));
