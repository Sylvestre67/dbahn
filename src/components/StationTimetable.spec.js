import React from 'react';
import { shallow } from 'enzyme';

import StationTimetable from './StationTimetable';
let component;

const props = {
	classes: { root: 'testing' },
};

describe('StationTimetable', () => {
	beforeAll(() => {
		component = shallow(<StationTimetable {...props} />);
	});

	it('should render correctly', () => {
		expect(component).toMatchSnapshot();
	});
});
