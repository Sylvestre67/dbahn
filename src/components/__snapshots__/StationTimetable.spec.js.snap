// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StationTimetable should render correctly 1`] = `
<withRouter(StationTimetable)
  classes={
    Object {
      "root": "withRouter-StationTimetable--root-1 testing",
    }
  }
/>
`;
